packages/snap/src/service.ts

```diff
@@ -55,9 +64,28 @@ export async function getLxpBalanceForAddress(
     return 0;
   }
   if (chainId === '0xe708') {
-    return getLxpBalanceFromChain(address);
+    return getTokenBalanceFromChain(LXP_CONTRACT_ADDRESS, address);
+  }
+  return getTokenBalanceFromLineascan(LXP_CONTRACT_ADDRESS, address);
+}
+
+/**
+ * Get the LXP-L balance for an address.
+ * @param address - The address to get the LXP-L balance for.
+ * @param chainId - The chain ID.
+ * @returns The LXP balance for the address.
+ */
+export async function getLxpLBalanceForAddress(
+  address: string,
+  chainId: string,
```

Comment:
Lxp balances are always on Linea Mainnet- so do we need to pass this as a params between methods ?
[Not a blocker]

Reply:
Yes, but depending on the network your MM is connected to, we fetch the balance via an RPC call or the Lineascan API. That's why we need it.